### ISSUE TITLE
Tests

### ISSUE THREAD

**Issue by simsurace** [JuliaStats/Distributions.jl#1573]
I think several things about the tests here are suboptimal and potentially confusing, making them hard to maintain.
- The directory/file structure in test/ does not match that of src/.
- It is unclear what is being tested. Are the tests more or less the same across distributions (it seems not)?
- The print output while running the tests is hard to read.
- There seems to be a dependency on R. Is this required or could the tests be fully in Julia?
- The coverage reports seem to be out-of-date (last build is from 2020?)

It seems that a general overhaul of tests would benefit this package. Is someone already working on this, or would an initial PR be welcome? Probably the overall task is best done by a team as it seems quite a commitment.

I would initially propose to get the directory structure in order, without really changing any tests, and fix the coverage reports. This will hopefully make it easier to understand what's going on. Next step after that could be to write a unified testing interface.

**Comment by devmotion** [comment]
Yeah, some things could really be improved. Eg I think it would be good to get rid of the evaluation of JSON files and use proper Julia files instead, and to add a proper test interface for univariate, multivariate etc. distributions. To me that feels even more useful than reorganizing the file structure (some traits are tested e.g. in common files but I guess that's OK for now - it's also simpler if at some point one wants to extract parts, like e.g. divergences or integrals). IIRC there are some open issues related to tests (or was it a PR?). I'll try to find them.

My general policy in the last months was to ensure that every addition or change is covered by tests, and use and extend the existing setup.

**Comment by devmotion** [comment]
E.g., [fit](https://github.com/JuliaStats/Distributions.jl/blob/master/test/fit.jl), [convolution](https://github.com/JuliaStats/Distributions.jl/blob/master/test/convolution.jl), [functionals](https://github.com/JuliaStats/Distributions.jl/blob/master/test/functionals.jl), [gradlogpdf](https://github.com/JuliaStats/Distributions.jl/blob/master/test/gradlogpdf.jl), and [pdf norm](https://github.com/JuliaStats/Distributions.jl/blob/master/test/pdfnorm.jl).

**Comment by matbesancon** [comment]
@JuliaRegistrator register
